src/yellowpaper/chapters/03-implementation/private-chain.tex

```diff
@@ -94,9 +94,10 @@ \subsection{Private chain}
 $\alpha_{pub}$ and $\beta_{pub}$ are parameters of the network -- a small
 constant fee and a linear price coefficient accordingly.
 
-To achieve the ability to prove the number of transactions in the Merkle tree,
-we will store it together with a hash in each node (as shown in
-\ref{fig:merkle-tree}). So every transaction disclosure will also verify the size.
+From the description above we can conclude that an educator may have an incentive to lie about the size of the tree. To achieve the ability to prove the number of transactions in the Merkle tree,
+we will store the size of the subtree with the hash in each node (as shown in \ref{fig:merkle-tree}). So every transaction disclosure will also verify the size. If an educator publishes digest of the tree with the wrong size, any transaction won't pass validation.
+
+Let's look at the example from the figure \ref{fig:merkle-tree}. Suppose we need to disclose $data[2]$. The path to the node H colored in gray. The proof nodes colored in yellow. So the size of the tree is $$C.size+D.size+I.size+1=5$$
```

Comment:
Why not add a couple of sentenses here like "Suppose that an Educator published a header hash with a wrong size, then every student which is collecting his own fair CV will see that an Educator deceives him, and moreover not a single data-disclosure deal would proceed", smth like this?